theano/tensor/signal/tests/test_conv.py

```diff
@@ -82,14 +82,13 @@ def sym_conv2d(input, filters):
 
     def test_basic(self):
         """
-        Basic functionality of nnet.conv.ConvOp is already tested by
-        its own test suite.  We just have to test whether or not
-        signal.conv.conv2d can support inputs and filters of type
-        matrix or tensor3.
+        Basic functionality of nnet.abstract_conv.AbstractConvOp is
+        already tested by its own test suite.  We just have to test
+        whether or not signal.conv.conv2d can support inputs and
+        filters of type matrix or tensor3.
         """
-        if(not theano.tensor.nnet.conv.imported_scipy_signal and
-           theano.config.cxx == ""):
-            raise SkipTest("conv2d tests need SciPy or a c++ compiler")
+        if(theano.config.cxx == "" or theano.config.mode == "FAST_COMPILE"):
+            raise SkipTest("conv2d tests need a c++ compiler")
```

Comment:
After adding the cxx_only flag, some test would raise error at FAST_COMPILE_NO_C. 
======================================================================
FAIL: theano.sparse.tests.test_opt.test_local_mul_s_d
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/pyenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/Theano/Theano/theano/sparse/tests/test_opt.py", line 81, in test_local_mul_s_d
    in f.maker.fgraph.toposort())
AssertionError
======================================================================
FAIL: theano.sparse.tests.test_opt.test_local_mul_s_v
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/pyenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/Theano/Theano/theano/sparse/tests/test_opt.py", line 99, in test_local_mul_s_v
    in f.maker.fgraph.toposort())
AssertionError
======================================================================
FAIL: theano.sparse.tests.test_opt.test_local_structured_add_s_v
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/pyenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/Theano/Theano/theano/sparse/tests/test_opt.py", line 117, in test_local_structured_add_s_v
    in f.maker.fgraph.toposort())
AssertionError
======================================================================
FAIL: theano.sparse.tests.test_opt.test_local_sampling_dot_csr
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/pyenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/Theano/Theano/theano/sparse/tests/test_opt.py", line 137, in test_local_sampling_dot_csr
    in f.maker.fgraph.toposort())
AssertionError